src/aggregator/aggregator/entry.go

```diff
@@ -1118,10 +1143,15 @@ func (e *Entry) addForwardedWithLock(
 	return err
 }
 
-func (e *Entry) shouldExpire(now xtime.UnixNano) bool {
+func (e *Entry) shouldExpire(
+	now xtime.UnixNano,
+	metricCategory metricCategory,
+) bool {
 	// Only expire the entry if there are no active writers
 	// and it has reached its ttl since last accessed.
-	return e.numWriters.Load() == 0 && now.After(xtime.UnixNano(e.lastAccessNanos.Load()).Add(e.opts.EntryTTL()))
+	age := now.Sub(xtime.UnixNano(e.lastAccessNanos.Load()))
+	e.metrics.entryExpiryByCategory[metricCategory].RecordDuration(age)
```

Comment:
I think you're going to double count entires. the `tick` method is weird and uses `ShouldExpire` to populate a set `entries` to expire with a call to `purgeExpired`, which then uses `TryExpire`. might need to refactor a bit.

Reply:
Ugh that's annoying, yeah will look into it

Reply:
Ah yeah I see the issue, good catch! Guess it wouldn't really be _too_ bad since these are hists so double counting should be equivalent to a single count, but will fix this up